plugins/utils/firenvim.nix

```diff
@@ -0,0 +1,86 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  types = lib.nixvim.nixvimTypes;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "firenvim";
+  defaultPackage = pkgs.vimPlugins.firenvim;
+
+  maintainers = with lib.maintainers; [ GaetanLepage ];
+
+  settingsOptions = {
+    globalSettings = defaultNullOpts.mkAttrsOf' {
+      type = types.anything;
+      pluginDefault = { };
+      example = {
+        alt = "all";
+      };
+      description = "Default settings that apply to all URLs.";
+    };
+
+    localSettings = defaultNullOpts.mkAttrsOf' {
+      type = with types; attrsOf anything;
+      pluginDefault = { };
+      example = {
+        ".*" = {
+          cmdline = "neovim";
+          content = "text";
+          priority = 0;
+          selector = "textarea";
+          takeover = "always";
+        };
+        "https?://[^/]+\\.co\\.uk/" = {
+          takeover = "never";
+          priority = 1;
+        };
+      };
+      description = ''
+        Map Javascript patterns that match against the full URL to settings that are used for all
+        URLs matched by that pattern.
+        When multiple patterns match a URL, the pattern with the highest "priority" value is used.
+      '';
+    };
+  };
+
+  settingsExample = {
+    globalSettings.alt = "all";
+    localSettings = {
+      ".*" = {
+        cmdline = "neovim";
+        content = "text";
+        priority = 0;
+        selector = "textarea";
+        takeover = "always";
+      };
+      "https?://[^/]+\\.co\\.uk/" = {
+        takeover = "never";
+        priority = 1;
+      };
+    };
+  };
+
+  settingsDescription = ''
+    Options fed to the `vim.g.firenvim_config` table.
```

Comment:
With this and lz-n, this is becoming a pattern. We should consider having a variant of or option in `mkVimPlugin` to handle these style of plugin configs.

Reply:
Yes, we could.
At the same time, adding yet another abstraction for such few occurrences does not seem to urgent to me.
For sure we can consider it a bit later.

Reply:
Yes, wasn't meant to be addressed urgently. Was just thinking if we keep coming across this style of settings.